presto-password-authenticators/src/main/java/io/prestosql/plugin/password/LdapAuthenticator.java

```diff
@@ -55,31 +55,51 @@
 {
     private static final Logger log = Logger.get(LdapAuthenticator.class);
 
-    private final String userBindSearchPattern;
+    private final Optional<String> userBindSearchPattern;
     private final Optional<String> groupAuthorizationSearchPattern;
     private final Optional<String> userBaseDistinguishedName;
+    private final Optional<String> bindDistinguishedName;
+    private final Optional<String> bindPassword;
     private final Map<String, String> basicEnvironment;
     private final LoadingCache<Credentials, Principal> authenticationCache;
 
     @Inject
     public LdapAuthenticator(LdapConfig serverConfig)
     {
         String ldapUrl = requireNonNull(serverConfig.getLdapUrl(), "ldapUrl is null");
-        this.userBindSearchPattern = requireNonNull(serverConfig.getUserBindSearchPattern(), "userBindSearchPattern is null");
+        this.userBindSearchPattern = Optional.ofNullable(serverConfig.getUserBindSearchPattern());
         this.groupAuthorizationSearchPattern = Optional.ofNullable(serverConfig.getGroupAuthorizationSearchPattern());
         this.userBaseDistinguishedName = Optional.ofNullable(serverConfig.getUserBaseDistinguishedName());
+        this.bindDistinguishedName = Optional.ofNullable(serverConfig.getBindDistingushedName());
+        this.bindPassword = Optional.ofNullable(serverConfig.getBindPassword());
+
         if (groupAuthorizationSearchPattern.isPresent()) {
             checkState(userBaseDistinguishedName.isPresent(), "Base distinguished name (DN) for user is null");
         }
+        checkState(bindDistinguishedName.isPresent() == bindPassword.isPresent(),
```

Comment:
code style: all arguments on one line or each on separate line
```suggestion
        checkState(
            bindDistinguishedName.isPresent() == bindPassword.isPresent(),
```